webapi/CopilotChat/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -424,21 +424,13 @@ public async Task<string> ExtractAudienceAsync(SKContext context)
     }
 
     /// <summary>
-    /// Helper function create the correct context variables to
-    /// extract audience from the conversation history.
+    /// Helper function that creates the correct context variables to
+    /// extract the audience from a conversation history.
     /// </summary>
     private async Task<string> GetAudienceAsync(SKContext context)
     {
-        var contextVariables = new ContextVariables();
-        contextVariables.Set("chatId", context["chatId"]);
-
-        var audienceContext = new SKContext(
-            contextVariables,
-            context.Memory,
-            context.Skills,
-            context.Log,
-            context.CancellationToken
-        );
+        SKContext audienceContext = context.Clone();
+        audienceContext.Variables.Set("chatId", context["chatId"]);
```

Comment:
If we are cloning the entire context, do we still need to set the variables? Same goes to other methods.

Reply:
Thanks for this.  Fixed.